torch/csrc/inductor/aoti_torch/shim_common.cpp

```diff
@@ -57,16 +57,24 @@ namespace fs = std::experimental::filesystem;
 
 #ifndef _WIN32
 #include <sys/stat.h>
+#include <sys/types.h> 
+#include <limits.h>
+#include <unistd.h>
 #endif
 
 // HACK for failed builds in ARVR, where it cannot find these symbols within
 // std::experimental::filesystem
 namespace {
-fs::path get_current_path() {
-#if __has_include("filesystem")
-  return fs::current_path();
+std::string get_current_path() {
+#ifdef _WIN32
+  return fs::current_path().string();
 #else
-  throw std::runtime_error("Not implemented");
+  char currentPath[PATH_MAX];
+  if (getcwd(currentPath, sizeof(currentPath)) != nullptr) {
+    return std::string(currentPath);
+  } else {
+    throw std::runtime_error("Failed to get current path");
```

Comment:
When this is possible?

Reply:
When your current working directory is deleted for example

Reply:
See https://godbolt.org/z/4on5Y6nsG